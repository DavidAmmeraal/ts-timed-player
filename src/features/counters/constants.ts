export const ADD = 'counters/ADD';
export const INCREMENT = 'counters/INCREMENT';
