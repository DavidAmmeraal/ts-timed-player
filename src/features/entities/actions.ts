/**
 * actions.ts
 * Action creators for entities actions.
 */
import Types from 'Types';
import { createAction } from 'typesafe-actions';

export const createEntityAction = createAction(
  'entities/CREATE_ENTITY',
  resolve => (entityType: string, props: Types.EntityProps) =>
    resolve({
      entityType,
      props,
    }),
);

export const updateEntityAction = createAction(
  'entities/UPDATE_ENTITY',
  resolve => (entityType: string, props: Types.EntityProps) =>
    resolve({
      entityType,
      props,
    }),
);

export const deleteEntityAction = createAction(
  'entities/DELETE_ENTITY',
  resolve => (entityType: string, id: string) =>
    resolve({
      entityType,
      id,
    }),
);
