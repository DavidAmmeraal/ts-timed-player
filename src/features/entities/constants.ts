export const CREATE = 'entities/CREATE';
export const UPDATE = 'entities/UPDATE';
export const DELETE = 'entities/DELETE';