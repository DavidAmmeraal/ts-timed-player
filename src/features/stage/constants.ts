export const FETCH_STAGE_REQUEST = 'stage/FETCH_STAGE_REQUEST';
export const FETCH_STAGE_SUCCESS = 'stage/FETCH_STAGE_SUCCESS';
export const FETCH_STAGE_ERROR = 'stage/FETCH_STAGE_ERROR';
export const HYDRATE_STAGE = 'stage/HYDRATE_STAGE';