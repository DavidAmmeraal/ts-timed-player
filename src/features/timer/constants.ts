/**
 * constants.ts
 * Constants for timer feature.
 */
export const TICK = 'timer/TICK';
export const TOGGLE = 'timer/TOGGLE';
export const SET_RATE = 'timer/SET_RATE';
export const SET_TIME = 'timer/SET_TIME';
