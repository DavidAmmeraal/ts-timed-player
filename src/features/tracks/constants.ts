export const TRACK_ENTITY_NAME = 'Track';