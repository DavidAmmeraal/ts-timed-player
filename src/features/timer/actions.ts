/**
 * actions.ts
 * Action creators for timer.
 */
import { action } from 'typesafe-actions';

import { TOGGLE, TICK, SET_RATE, SET_TIME } from './constants';

export const toggle = () => action(TOGGLE);
export const tick = (time: number) => action(TICK, time);
export const setRate = (rate: number) => action(SET_RATE, rate);
/**
 * Creates an SET_TIME time action.
 * @param time time in microseconds
 */
export const setTime = (time: number) => action(SET_TIME, time);
