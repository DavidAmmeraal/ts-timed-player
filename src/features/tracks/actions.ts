import { action } from 'typesafe-actions';

import { CREATE_TRACK, UPDATE_TRACK, DELETE_TRACK } from './constants';
import { ITrack } from './models';

export const createTrack = (track:ITrackEntity) => action(CREATE_TRACK, track);
export const updateTrack = (id:string, track: Partial<ITrack>) => action(UPDATE_TRACK, {
    id,
    track,
});
export const deleteTrack = (id:string) => action(DELETE_TRACK, id);
