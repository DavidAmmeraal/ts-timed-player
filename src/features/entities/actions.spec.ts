import Types from 'Types';
/**
 * actions.spec.ts
 * Unit tests entities action creators.
 */
import * as actions from './actions';
import * as constants from './constants';

interface TestEntityProps extends Types.EntityProps {
  foo: string,
  bar: string,
}

type TestEntity = Types.Entity<'TestEntity', TestEntityProps>;

describe('createEntityAction()', () => {
  it('should create a new CREATE action.', () => {
    const props = { foo: 'bar', bar: 'foo' };
    const id = '1';
    const type = 'TestEntity'; 
    const action = actions.createEntityAction<TestEntity>({type, id, props});
    expect(action).toMatchObject({
      type: constants.CREATE_ENTITY,
      payload: {
        id,
        type,
        props,
      }
    })
  });
});

describe('updateEntityAction()', () => {
  it('should create a new UPDATE action.', () => {
    const action = actions.updateEntityAction<TestEntity>({type: 'TestEntity', id: '1', props: { bar: 'bar' }});
    expect(action).toMatchSnapshot();
  });
});

describe('deleteEntityAction()', () => {
  it('should create a new DELETE action.', () => {
    const action = actions.deleteEntityAction('X', '123');
    expect(action).toMatchSnapshot();
  });
});
