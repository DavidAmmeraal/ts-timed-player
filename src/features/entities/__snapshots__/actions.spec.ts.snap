// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createEntityAction() should create a new CREATE action. 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "entity": Object {
      "foo": "bar",
      "id": "1",
    },
    "entityType": "TestEntity",
  },
  "type": "entities/CREATE",
}
`;

exports[`deleteEntityAction() should create a new DELETE action. 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "entityType": "X",
    "id": "123",
  },
  "type": "entities/DELETE",
}
`;

exports[`updateEntityAction() should create a new UPDATE action. 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "entityType": "X",
    "id": "123",
    "props": Object {
      "foo": "bar",
    },
  },
  "type": "entities/UPDATE",
}
`;
